### DISCUSSION TITLE
Server action for a button w/o making entire form - like rails button_to "Text", some_server_path, method: :post

### DISCUSSION THREAD

**Discussion by sethhorsley** [vercel/next.js#60643] [Ideas]
### Goals

1. Add functionality and flexibility to server actions for single action button server actions
2. Simplify very simple form logic

### Background

Being a rails / js developer next js could use the approach that rails has with their button_to with post functionality without the need to scaffold out an entire form like:
```ruby
button_to "Button Text", some_server_path, method: :post
```

### Proposal

Not entirely sure how this should be implemented, just wanted to start a dialog and see where this goes.

**Comment by nirvikpurkait** [comment]
I think you can make use of `formAction` prop on button just making the `button` wrap by `from` tag, make sure the button has `type="submit"` as attribute.

Or 

You can expose the button as client component, add an `onClick` event listener and `await` the server-action, make user the server-action is in a different file and it has `"use server"` at the top.
